src/credentials/activation.ts

```diff
@@ -5,15 +5,42 @@
 
 import * as vscode from 'vscode'
 import { AwsContext } from '../shared/awsContext'
+import { profileSettingKey } from '../shared/constants'
 import { CredentialsProfileMru } from '../shared/credentials/credentialsProfileMru'
+import { UserCredentialsUtils } from '../shared/credentials/userCredentialsUtils'
+import { SettingsConfiguration } from '../shared/settingsConfiguration'
 
 export interface CredentialsInitializeParameters {
     extensionContext: vscode.ExtensionContext
     awsContext: AwsContext
+    settingsConfiguration: SettingsConfiguration
 }
 
 export async function initialize(parameters: CredentialsInitializeParameters): Promise<void> {
     updateMruWhenAwsContextChanges(parameters.awsContext, parameters.extensionContext)
+    updateConfigurationWhenAwsContextChanges(
```

Comment:
Do we need these as separate functions if they're only used here?

Reply:
I thought about this. I figured the initial implementation should keep them separate so that they can be torn away, and it keeps the logic in each handler simple. We can revisit over time if we feel one bigger handler makes sense.